cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/BaseController.java

```diff
@@ -114,22 +115,28 @@ public static void main(String[] args) {
             } else if (ex instanceof ApplyException) {
                 log.error("An error occurred during the apply:", ex.getMessage());
             } else if (ex instanceof GetException) {
-                log.error("An error occurred during the get:", ex.getMessage());
-            } else if (ex instanceof IllegalStateException) {
-                // a little bit ugly, but it works
-                // the problem is in reactor.core.Exceptions
-                // it creates lambda class instances which are hard to test on...
-                if (ex.getMessage().toLowerCase().contains("retries exhausted")) {
-                    // now, if there is a cause and it's an InvalidTokenException, we can at least provide a hint
-                    // that it's likely a password issue
-                    if (ex.getCause() != null &&
-                            ex.getCause().toString().toLowerCase().contains("invalidtokenexception")) {
-                        log.error("Request to CF API failed: invalid token error (is the password correct?)");
-                    } else {
-                        log.exception(ex, "Request to CF API failed:");
-                    }
+                Throwable getExceptionCause = ex.getCause();
+
+                // wrapped unknown host exceptions stand for unreachable hosts
+                if (getExceptionCause.getCause() instanceof UnknownHostException) {
+                    log.error("Unable to connect to the CF API host:", getExceptionCause.getMessage());
+
+                    // illegal argument exceptions seem to denote invalid organizations and spaces
```

Comment:
This comment layout is really awkward to read. Consider moving the comments which explain the `if` structure above the entire construct.

Reply:
I try to explain the if structure with an additional comment now.